src_py/sysfont.py

```diff
@@ -58,26 +58,35 @@ def initsysfonts_win32():
     fonts = {}
 
     # add fonts entered in the registry
-    key_path = "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Fonts"
-    key = _winreg.OpenKey(_winreg.HKEY_LOCAL_MACHINE, key_path)
-
-    for i in range(_winreg.QueryInfoKey(key)[1]):
-        try:
-            # name is the font's name e.g. Times New Roman (TrueType)
-            # font is the font's filename e.g. times.ttf
-            name, font, _ = _winreg.EnumValue(key, i)
-        except OSError:
-            break
-
-        if splitext(font)[1].lower() not in OpenType_extensions:
-            continue
-        if not dirname(font):
-            font = join(fontdir, font)
+    MICROSOFT_FONT_DIRS = [
+        "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Fonts",
+        "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Fonts",
+    ]
```

Comment:
Replacing the highlighted code with the new one causes the `match_font()` method to not return anything. It was working fine before:
```python
PS C:\users\novia\documents\github\pygame> python
Python 3.9.0 (tags/v3.9.0:9cf6752, Oct  5 2020, 15:34:40) [MSC v.1927 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import pygame
pygame 2.1.3.dev5 (SDL 2.0.22, Python 3.9.0)
Hello from the pygame community. https://www.pygame.org/contribute.html
>>> pygame.font.match_font("gothambold")
'C:\\Users\\novia\\AppData\\Local\\Microsoft\\Windows\\Fonts\\gothambold.ttf'
>>> pygame.font.match_font("arial")
'C:\\WINDOWS\\Fonts\\arial.ttf'
```

Reply:
I was suggesting to use the user paths only in the condition that `domain` is `_winreg.HKEY_CURRENT_USER` in the loop. Global fonts (when domain is `_winreg.HKEY_LOCAL_MACHINE`) should still use global path prefixes

I'm surprised the global path works for both cases in the loop (does it though?)

Reply:
Yeah, apparently it does work for some reason. I'm not so sure if the current fix is a monkey patch or not.

Reply:
I think this should be fixed for the sake of extra safely, even though it seems to work for now.

Reply:
Okay, on it.

Reply:
To recap what we have said over Discord, apparently using the user paths that @ankith26 has specified does not work for some sus reason.

Reply:
Here is the code that does not work: https://www.toptal.com/developers/hastebin/ahozifilap.py

Reply:
They aren't paths, despite the variable name. Novial's PR works, and it works because the paths are paths to registry keys, and they are correct. You can verify this on your system by going into the "Registry Editor" application. For me, these are the two valid key paths:

```
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Fonts
HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Fonts
```